clang/lib/Sema/SemaTemplateVariadic.cpp

```diff
@@ -262,20 +263,34 @@ namespace {
       if (!Lambda->containsUnexpandedParameterPack())
         return true;
 
-      bool WasInLambda = InLambda;
+      bool WasInLambdaOrBlock = InLambdaOrBlock;
       unsigned OldDepthLimit = DepthLimit;
 
-      InLambda = true;
+      InLambdaOrBlock = true;
       if (auto *TPL = Lambda->getTemplateParameterList())
         DepthLimit = TPL->getDepth();
 
       inherited::TraverseLambdaExpr(Lambda);
 
-      InLambda = WasInLambda;
+      InLambdaOrBlock = WasInLambdaOrBlock;
       DepthLimit = OldDepthLimit;
       return true;
     }
 
+    /// Analogously for blocks.
+    bool TraverseBlockExpr(BlockExpr *Block) {
+      if (!Block->containsUnexpandedParameterPack())
+        return true;
+
+      bool WasInLambdaOrBlock = InLambdaOrBlock;
+      InLambdaOrBlock = true;
+
+      inherited::TraverseBlockExpr(Block);
+
+      InLambdaOrBlock = WasInLambdaOrBlock;
```

Comment:
Maybe use llvm::SaveAndRestore?

Reply:
*That’s* what it was called. When I looked at the code that handles lambdas above I was like ‘don’t we have an RAII thingy for this?’ but I couldn’t remember what it was called, so thanks!